packages/code-studio/src/settings/EditorSectionContent.tsx

```diff
@@ -51,15 +51,24 @@ export function EditorSectionContent(): JSX.Element {
     (newValue: Record<string, unknown>) => {
       dispatch(
         updateNotebookSettings({
-          python: { linter: { ...ruffSettings, config: newValue } },
+          python: {
+            linter: {
+              ...ruffSettings,
+              config:
+                JSON.stringify(newValue) ===
+                JSON.stringify(RUFF_DEFAULT_SETTINGS)
+                  ? undefined
+                  : newValue,
+            },
+          },
         })
       );
       MonacoProviders.setRuffSettings(newValue);
     },
     [dispatch, ruffSettings]
   );
 
-  const [val, setVal] = useState(JSON.stringify(ruffConfig, null, 2));
+  const val = JSON.stringify(ruffConfig, null, 2);
```

Comment:
Should we memoize this?

Reply:
I don't think it matters. The object isn't that big and there's still some overhead to do the string comparison for memoization. Also the component shouldn't re-render often.